js/ui/html_editor/utils/templates_storage.js

```diff
@@ -0,0 +1,39 @@
+import { isDefined } from '../../../core/utils/type';
+
+export default class TemplatesStorage {
+    constructor() {
+        this._storage = {};
+    }
+
+    set({ widgetKey, marker }, value) {
+        this._storage[widgetKey] ??= {};
+        this._storage[widgetKey][marker] = value;
+    }
+
+    get({ widgetKey, marker }) {
+        // reason for add double key is T1110266
+        if(isDefined(widgetKey)) {
+            const widgetTemplates = this._storage[widgetKey];
+
+            return widgetTemplates ? widgetTemplates[marker] : undefined;
+        } else {
+            // When Quill parse markup, we do not have info about widget (widgetKey)
+            // In this case, we take latest template from storage because this (latest) template related to current widget(this widget initialized parsing of markup)
```

Comment:
We usually use a `NOTE:` prefix for such comments.
```javascript
            // NOTE: If anonymous templates is used, mention is parsed from the markup.
            // Quill format does not have information about related HtmlEditor instance.
            // In this case, the latest template in the storage is what we need
            // because appropriate instance has already been created and has added its templates to the storage.
```

Please, review it with redactors

Reply:
checked